ppr-ui/src/utils/registration-helper.ts

```diff
@@ -22,6 +24,147 @@ import {
+  if (courtOrder && courtOrder.courtName !== '' && courtOrder.courtRegistry !== '' &&
+      courtOrder.fileNumber !== '' && courtOrder.effectOfOrder !== '' && courtOrder.orderDate !== '') {
+    statement.changeType = APIAmendmentTypes.COURT_ORDER
+    statement.courtOrderInformation = stateModel.registration.courtOrderInformation
+  } else {
+    statement.changeType = APIAmendmentTypes.AMENDMENT
+    delete statement.courtOrderInformation
+  }
+  // Set these every time.
+  statement.addSecuredParties = []
+  statement.deleteSecuredParties = []
+  statement.addDebtors = []
+  statement.deleteDebtors = []
+  statement.addVehicleCollateral = []
+  statement.deleteVehicleCollateral = []
+  statement.addGeneralCollateral = []
+  statement.deleteGeneralCollateral = []
+  if (stateModel.folioOrReferenceNumber && stateModel.folioOrReferenceNumber !== '') {
+    statement.clientReferenceId = stateModel.folioOrReferenceNumber
+  } else {
+    statement.clientReferenceId = ''
+  }
+
+  // trust indenture setup
+  if (stateModel.registration.lengthTrust.action &&
+      stateModel.registration.lengthTrust.action === ActionTypes.EDITED &&
+      registrationType.registrationTypeAPI === APIRegistrationTypes.SECURITY_AGREEMENT) {
+    statement.trustIndenture = stateModel.registration.lengthTrust.trustIndenture
+  } else {
+    statement.trustIndenture = false
+  }
+  const parties:AddPartiesIF = stateModel.registration.parties
+  // Conditionally set draft debtors.
+  for (let i = 0; i < parties.debtors.length; i++) {
+    if (parties.debtors[i].action) {
+      if (parties.debtors[i].action === ActionTypes.ADDED || parties.debtors[i].action === ActionTypes.EDITED) {
+        statement.addDebtors.push(JSON.parse(JSON.stringify(parties.debtors[i])))
+      }
+      if (parties.debtors[i].action === ActionTypes.REMOVED || parties.debtors[i].action === ActionTypes.EDITED) {
+        statement.deleteDebtors.push(JSON.parse(JSON.stringify(parties.debtors[i])))
+      }
+    }
+  }
+  // Conditionally set draft secured parties.
+  for (let i = 0; i < parties.securedParties.length; i++) {
+    if (parties.securedParties[i].action) {
+      if (parties.securedParties[i].action === ActionTypes.ADDED ||
+          parties.securedParties[i].action === ActionTypes.EDITED) {
+        statement.addSecuredParties.push(JSON.parse(JSON.stringify(parties.securedParties[i])))
+      }
+      if (parties.securedParties[i].action === ActionTypes.REMOVED ||
+          parties.securedParties[i].action === ActionTypes.EDITED) {
+        statement.deleteSecuredParties.push(JSON.parse(JSON.stringify(parties.securedParties[i])))
+      }
+    }
+  }
+
+  const collateral:AddCollateralIF = stateModel.registration.collateral
+  // Conditionally set draft vehicle collateral
+  if (collateral.vehicleCollateral && collateral.vehicleCollateral.length > 0) {
+    for (let i = 0; i < collateral.vehicleCollateral.length; i++) {
```

Comment:
hey doug, do you think you could make this for loop into it's own fn and use it for parties/debtors as well? Maybe pass in the the list your iterating, the statement.add... list and the statement.delete.. list. I can see someone coming in here to edit something later in ops and messing it up with a copy/paste error haha

Reply:
Good idea, added function in second commit.